crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -375,64 +374,38 @@ impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientState {
                 let key_utf8 = keymap_state.key_get_utf8(Keycode::from(key + MIN_KEYCODE));
                 let key_sym = keymap_state.key_get_one_sym(Keycode::from(key + MIN_KEYCODE));
 
-                let key = if matches!(
-                    key_sym,
-                    xkb::Keysym::BackSpace
-                        | xkb::Keysym::Left
-                        | xkb::Keysym::Right
-                        | xkb::Keysym::Down
-                        | xkb::Keysym::Up
-                        | xkb::Keysym::Super_L
-                        | xkb::Keysym::Super_R
-                ) {
-                    xkb::keysym_get_name(key_sym).to_lowercase()
-                } else {
-                    key_utf8.clone()
-                };
+                state.modifiers.shift =
+                    keymap_state.mod_name_is_active(xkb::MOD_NAME_SHIFT, xkb::STATE_MODS_EFFECTIVE);
+                state.modifiers.alt =
+                    keymap_state.mod_name_is_active(xkb::MOD_NAME_ALT, xkb::STATE_MODS_EFFECTIVE);
+                state.modifiers.control =
+                    keymap_state.mod_name_is_active(xkb::MOD_NAME_CTRL, xkb::STATE_MODS_EFFECTIVE);
+                state.modifiers.command =
+                    keymap_state.mod_name_is_active(xkb::MOD_NAME_LOGO, xkb::STATE_MODS_EFFECTIVE);
```

Comment:
I think that there is no need to store modifiers anymore and it's better to add `get_modifiers()` to `WaylandWindowState` and use `state.get_modifiers()` instead of `state.modifiers` everywhere.

Reply:
True, `modifiers` isn't needed anymore so we can reduce the code. I've pushed a commit that keeps `state.modifiers` and only updates it when we get the `modifiers` event. Should keep us from having to call these C functions repeatedly.